pkg/blockstorage/azure/azuredisk.go

```diff
@@ -420,16 +463,27 @@ func (s *AdStorage) snapshotParse(ctx context.Context, snap armcompute.Snapshot)
 	if snap.Tags != nil {
 		tags = blockstorage.StringMap(snap.Tags)
 	}
+
+	diskSize := azto.Ptr(int64(0))
```

Comment:
@pavannd1 @viveksinghggits should  `diskSize` be considered critical data and results in an error if empty?

Reply:
hmm, 0 bytes of snapshot doesn't really make sense, but I am not really sure. Let's see what Pavan has to say.

Reply:
```
	// If creationData.createOption is Empty, this field is mandatory and it indicates the size of the disk to create. If this
	// field is present for updates or creation with other options, it indicates a
	// resize. Resizes are only allowed if the disk is not attached to a running VM, and can only increase the disk's size.
	DiskSizeGB *[int32](https://pkg.go.dev/builtin#int32)
```
Seems okay to me based on this description. It's not a mandatory field unless the snapshot was created with `Empty` option.